flows/contact.go

```diff
@@ -11,17 +11,32 @@ import (
 
 // Contact represents a single contact
 type Contact struct {
-	uuid     ContactUUID
-	name     string
-	language utils.Language
-	timezone *time.Location
-	urns     URNList
-	groups   GroupList
-	fields   *Fields
+	uuid              ContactUUID
+	name              string
+	language          utils.Language
+	backdownLanguages utils.LanguageList
+	timezone          *time.Location
+	urns              URNList
+	groups            GroupList
+	fields            *Fields
 }
 
+// SetLanguage sets the language for this contact
 func (c *Contact) SetLanguage(lang utils.Language) { c.language = lang }
-func (c *Contact) Language() utils.Language        { return c.language }
+
+// Language gets the language for this contact
+func (c *Contact) Language() utils.Language { return c.language }
+
+// PreferredLanguages gets all languages for this contact in order of preference
+func (c *Contact) PreferredLanguages() utils.LanguageList {
+	var languages utils.LanguageList
+
+	if c.language != utils.NilLanguage {
+		languages = append(languages, c.language)
+	}
+
+	return append(languages, c.backdownLanguages...)
+}
```

Comment:
Say a contact starts with `'eng'` as a language, but backdown languages is set to `['fra', 'spa']`, if an update contact action setting the language to 'spa' happens, won't preferred languages return `['spa', 'fra', 'spa']`?